src/components/ScoreByGameId.js

```diff
@@ -1,42 +1,59 @@
 import React from "react";
 import Requests from "./Requests";
-import "../styles/scores.css"
+import "../styles/scores.css";
+
+// import Scores from "../components/Scores";
 
 class ScoresByGameId extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      scores: []
+    constructor(props) {
+        super(props);
+        this.state = {
+            id: '',
+            scores: []
+        };
+    }
+
+
+    async componentDidMount() {
+        const id = this.props.match.params.id;
+        this.setState({id, scores: await Requests.get("/scores/" + id)});
+    }
+
+    async componentDidUpdate(nextProps, prevState) {
+        const id = this.props.match.params.id;
+        if (id !== prevState.id) {
+            this.setState({id, scores: await Requests.get("/scores/" + id)});
```

Comment:
Please add the logic extract the setState to a method that's called from componentDidMount and componentDidUpdate